pagopa/io-app
# chore: [IOPID-2137,IOPID-2479] CieID events integration

## Short description
This PR adds event tracking to the `CieID` login flow and edit copy on `CieIdWizard` screen

## List of changes proposed in this pull request
- Added event tracking for the various scenarios in the new `CieID` login flow.
- Updated some `SPID` error events with the addition of some event `properties`
- Updated `CieIdWizard` description

## Demo
<details><summary>Test wizards</summary>
<video src="https://github.com/user-attachments/assets/5f2469b5-5eae-45f7-8874-e5da3e61e0b7"></video>
</details>
<details><summary>Test operation cancel and login success</summary>
<video src="https://github.com/user-attachments/assets/506e0a29-2720-46c2-b21b-aaa4e908cc4c"></video>
</details>

## Screen

<img width=300 src="https://github.com/user-attachments/assets/05edcd72-ac6c-4361-bbb4-28cabdf89153" />

## How to test
Run the app and navigate through the `CieID` login/wizards flows and check if all the events are properly dispatched.

### Events table
<details>

| TAG       | EVENT NAME                             | EVENT CATEGORY     | EVENT TYPE   | TRIGGER                                                           | PROPERTY 1               | PROPERTY 2               |
|:----------|:---------------------------------------|:-------------------|:-------------|:------------------------------------------------------------------|:-------------------------|:-------------------------|
|New        | LOGIN_CIE_IDENTIFICATION_MODE          | UX                 | screen view  | screen view > bottom sheet “How do you want to login?”            |                          |                          |
|New        | LOGIN_CIE_WIZARD_SELECTED              | UX                 | action       | tap > "We give you a hand”                                        |                          |                          |
|New        | LOGIN_CIE_WIZARD_CIEID                 | UX                 | screen view  | screen view > “Do you have the CieID app?”                        |                          |                          |
|New        | LOGIN_CIE_WIZARD_CIEID_SELECTED        | UX                 | action       | tap > "Yes, log in with CieID"                                    |                          |                          |
|New        | LOGIN_CIE_WIZARD_PIN                   | UX                 | screen view  | screen view > “Do you have your Electronic Identity Card and PIN?”|                          |                          |
|New        | LOGIN_CIE_WIZARD_PIN_INFO              | UX                 | action       | tap > "Where can I find the CIE PIN?"                             |                          |                          |
|New        | LOGIN_CIE_WIZARD_PIN_SELECTED          | UX                 | action       | tap > "Yes, enter with CIE + PIN"                                 |                          |                          |
|New        | LOGIN_CIE_WIZARD_SPID                  | UX                 | screen view  | screen view > “Do you have SPID?”                                 |                          |                          |
|New        | LOGIN_CIE_WIZARD_SPID_SELECTED         | UX                 | action       | tap > "Yes, login with SPID"                                      |                          |                          |
|New        | LOGIN_CIE_WIZARD_IDP_ACTIVATION        | UX                 | screen view  | screen view > "Activate your digital identity!"                   |                          |                          |
|New        | LOGIN_CIE_PIN_SELECTED                 | UX                 | action       | tap > "Login with CIE + PIN"                                      |                          |                          |
|New        | LOGIN_CIEID_SELECTED                   | UX                 | action       | tap > "Login with CieID"                                          | security_level = L2 / L3 |                          |
|New        | LOGIN_CIEID_APP_NOT_FOUND              | KO                 | error        | screen > "We can't find the CieID app"                            |                          |                          |
|New        | LOGIN_CIEID_APP_NOT_FOUND_DOWNLOAD     | UX                 | action       | tap > "scarica CieID"                                             |                          |                          |
|New        | LOGIN_CIEID_UX_SUCCESS                 | UX                 | confirm      | Login with CieID success                                          | login_session= 365 / 30  |                          |
|New        | LOGIN_ERROR_MESSAGE                    | KO                 | error        | screen view > "It was not possible to access"                     | idp                      |error message             |
|New        | LOGIN_CIEID_FALLBACK_CIE_PIN           | KO                 |              | screen > "Can't log in with CieID?"                               |                          |                          |
|New        | LOGIN_CIEID_FALLBACK_CIE_PIN_SELECTED  | UX                 | action       | tap > "Login with CIE + PIN"                                      |                          |                          |
|New        | LOGIN_CIEID_FALLBACK_SPID              | KO                 |              | screen > "Can't log in with CieID?"                               |                          |                          |
|New        | LOGIN_CIEID_FALLBACK_CIE_SPID_SELECTED | UX                 | action       | tap > "Login with SPID"                                           |                          |                          |
|Updated    | LOGIN_SPID_GENERIC_ERROR               | KO                 |              | event > "Generic error"                                           | idp                      | error message            |
|Updated    | LOGIN_SPID_ATTEMPTS_ERROR              | KO                 |              | event > "Max login attempts reached"                              | idp                      |                          |
|Updated    | LOGIN_SPID_SECURITY_LEVEL              | KO                 |              | event > "Security level"                                          | idp                      |                          |
|Updated    | LOGIN_SPID_TIMEOUT_ERROR               | KO                 |              | event > "Timeout error"                                           | idp                      |                          |
|Updated    | LOGIN_SPID_DATA_SHARING_ERROR          | KO                 |              | event > "Data sharing consent denied"                             | idp                      |                          |
|Updated    | LOGIN_SPID_IDENTITY_ERROR              | KO                 |              | event > "Identity suspended or revoked"                           | idp                      |                          |
|Updated    | LOGIN_SPID_CANCEL_ERROR                | KO                 |              | event > "login cancelled by the user"                             | idp                      |                          |

</details>
